tests/integration/test_custom_application_source_version.py

```diff
@@ -0,0 +1,231 @@
+    p.write_text(app_py)
+    os.utime(p, (ts, ts))
+    p = folder_path / "start-app.sh"
+    p.write_text(start_script)
+    os.utime(p, (ts, ts))
+    return str(folder_path.resolve())
+
+
+@pytest.fixture
+def additional_metadata_context_path(tmp_path: pathlib.Path, metadata: str) -> str:
+    ts = 1701797283
+    folder_path = tmp_path / "additional_app"
+    folder_path.mkdir()
+    p = folder_path / "metadata.yaml"
+    p.write_text(metadata)
+    os.utime(p, (ts, ts))
+    return str(folder_path.resolve())
+
+
+@pytest.fixture
+def custom_application_source(
+    dr_endpoint: str, dr_token: str, cleanup_dr: Any
+) -> Generator[str, None, None]:
+    with cleanup_dr("customApplicationSources/"):
+        yield get_or_create_custom_application_source(
+            endpoint=dr_endpoint,
+            token=dr_token,
+            name="pytest custom application source",
+        )
+
+
+@pytest.fixture
+def cleanup_env_ver(cleanup_dr: Any, custom_application_source: str) -> Generator[None, None, None]:
+    with cleanup_dr(f"customApplicationSources/{custom_application_source}/versions"):
+        yield
+
+
+def test_get_or_create(
+    dr_endpoint: str,
+    dr_token: str,
+    cleanup_env_ver: Any,
+    custom_application_source: str,
+    app_context_path: str,
+    additional_metadata_context_path: str,
+    credentials: str,
+) -> None:
+    client = dr.Client(endpoint=dr_endpoint, token=dr_token)  # type: ignore
+    env_ver_id_1 = get_or_create_custom_application_source_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_application_source_id=custom_application_source,
+        folder_path=app_context_path,
+        label="pytest app source version",
+    )
+    assert len(env_ver_id_1)
+
+    env_ver_id_2 = get_or_create_custom_application_source_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_application_source_id=custom_application_source,
+        folder_path=app_context_path,
+        label="pytest app source version",
+    )
+    assert env_ver_id_1 == env_ver_id_2
+
+    env_ver_id_3 = get_or_create_custom_application_source_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_application_source_id=custom_application_source,
+        folder_path=app_context_path,
+        label="pytest app source version alt",
+    )
+    assert env_ver_id_1 != env_ver_id_3
+    folder_contents = [
+        item["fileName"]
+        for item in client.get(
+            f"customApplicationSources/{custom_application_source}/versions/{env_ver_id_3}"
+        ).json()["items"]
+    ]
+    env_ver_id_4 = get_or_create_custom_application_source_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_application_source_id=custom_application_source,
+        previous_custom_application_source_version_id=env_ver_id_3,
+        folder_path=additional_metadata_context_path,
+    )
+    assert env_ver_id_3 == env_ver_id_4
```

Comment:
see my comment above - this behavior feels idiosyncratic and doesn't align to our precedent in custom model versions

Reply:
ditto to the below assertions on the id staying the same